src/cdk/v2/destinations/http/utils.js

```diff
@@ -41,12 +42,60 @@ const getCustomMappings = (message, mapping) => {
   }
 };
 
-const addPathParams = (message, apiUrl) => {
+const encodeParamsObject = (params) => {
+  if (!params || typeof params !== 'object') {
+    return {}; // Return an empty object if input is null, undefined, or not an object
+  }
+  return Object.keys(params)
+    .filter((key) => params[key] !== undefined)
+    .reduce((acc, key) => {
+      acc[encodeURIComponent(key)] = encodeURIComponent(params[key]);
+      return acc;
+    }, {});
+};
+
+const getPathValueFromJsonpath = (message, path) => {
+  let finalPath = path;
+  if (path.includes('/')) {
+    throw new ConfigurationError('Path value cannot contain "/"');
+  }
+  if (path.includes('$')) {
+    try {
+      [finalPath = null] = jsonpath.query(message, path);
+    } catch (error) {
+      throw new ConfigurationError(
+        `An error occurred while querying the JSON path: ${error.message}`,
+      );
+    }
+    if (finalPath === null) {
+      throw new ConfigurationError('Path not found in the object.');
+    }
+  }
+  return finalPath;
+};
+
+const getPathParamsSubString = (message, pathParamsArray) => {
+  if (pathParamsArray.length === 0) {
+    return '';
+  }
+  const pathParamsValuesArray = pathParamsArray.map((pathParam) =>
+    encodeURIComponent(getPathValueFromJsonpath(message, pathParam.path)),
+  );
+  return `/${pathParamsValuesArray.join('/')}`;
+};
+
+const prepareEndpoint = (message, apiUrl, pathParams) => {
+  let requestUrl;
   try {
-    return applyJSONStringTemplate(message, `\`${apiUrl}\``);
+    requestUrl = applyJSONStringTemplate(message, `\`${apiUrl}\``);
```

Comment:
requestUrl will be a constant string, why to process through jsontemplate here ?

Reply:
I don't remember discussing the feature for requestUrl to be dynamic based on event but will be good addition